File: eboto_hardhat/ignition/modules/deploy.ts
import { buildModule } from "@nomicfoundation/hardhat-ignition/modules"

export default buildModule("eBoto", (m) => {
    const authority_address = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
    const authority_pubkey = "8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed753547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5"
    const ea_account = m.contract("EA_Account", [authority_address, authority_pubkey]);
    return { ea_account };
});

File: eboto_hardhat/deploy.py
from sys import argv
import subprocess
from time import sleep

#Start the Hardhat Node
start_node_command = "npx hardhat node"
node_instance = subprocess.Popen(start_node_command.split())
print("Blockchain Started")

sleep(10)

#Erase the old history
erase_history_command = "rm -rf ignition/deployments/chain-31337"
subprocess.run(erase_history_command.split())


#Deploy the contract
deploy_contract_command = f"npx hardhat ignition deploy ignition/modules/{argv[1]} --network localhost"
subprocess.run(deploy_contract_command.split())

print("Contract Deployed")

#Generate the abis
abi_command = "npx solc@0.8.19 --abi contracts/EA_Account.sol"
subprocess.run(abi_command.split())




node_instance.wait()
